service/src/main/java/bio/terra/cbas/controllers/PublicApiController.java

```diff
@@ -1,16 +1,43 @@
 package bio.terra.cbas.controllers;
 
 import bio.terra.cbas.api.PublicApi;
+import bio.terra.cbas.config.CromwellServerConfiguration;
 import bio.terra.cbas.model.SystemStatus;
+import bio.terra.cbas.model.SystemStatusSystems;
+import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.http.HttpStatus;
 import org.springframework.http.ResponseEntity;
 import org.springframework.stereotype.Controller;
+import org.springframework.web.client.RestTemplate;
 
 @Controller
 public class PublicApiController implements PublicApi {
 
+  private final CromwellServerConfiguration cromwellConfig;
+
+  @Autowired
+  public PublicApiController(CromwellServerConfiguration cromwellConfig) {
+    this.cromwellConfig = cromwellConfig;
+  }
+
   @Override
   public ResponseEntity<SystemStatus> getStatus() {
-    return new ResponseEntity<>(new SystemStatus().ok(true), HttpStatus.OK);
+
+    RestTemplate restTemplate = new RestTemplate();
+    String result;
+    Boolean isOk;
```

Comment:
Two things; if you have a local var that will only be assigned once, you can make it clear to the reader (and compiler) by using `final`. Second, I would avoid using boxed types (like `Boolean`) unless you need to store `null` at some point. IMO this also makes the code's intent clearer.
```suggestion
    final String result;
    final boolean isOk;
```

Reply:
I am in support of "assign exactly once" style types, thanks for the suggestion.

I think the compiler might not be smart enough to understand that only one branch of the `try` will execute, getting `variable isOk might already have been assigned` in the `catch`.